tests/Doctrine/Tests/DBAL/Platforms/AbstractSQLServerPlatformTestCase.php

```diff
@@ -165,14 +167,18 @@ public function getGenerateForeignKeySql()
 
     public function testModifyLimitQuery()
     {
-        $sql = $this->_platform->modifyLimitQuery('SELECT * FROM user', 10, 0);
-        $this->assertEquals('SELECT * FROM (SELECT *, ROW_NUMBER() OVER (ORDER BY (SELECT 0)) AS doctrine_rownum FROM user) AS doctrine_tbl WHERE doctrine_rownum BETWEEN 1 AND 10 ORDER BY doctrine_rownum', $sql);
+        $querySql = 'SELECT * FROM user';
+        $alteredSql = 'SELECT TOP 10 * FROM user';
+        $sql = $this->_platform->modifyLimitQuery($querySql, 10, 0);
+        $this->assertEquals(sprintf(self::$selectFromCtePattern, $alteredSql, 1, 10), $sql);
```

Comment:
Theses tests are rather hard to read. I wonder if it would make sense to add our own assertion here like:

```php

protected function assertModifiesLimitQuery($query, $modifiedQuery, $limit, $offset)
{
    $selectFromCtePattern = "WITH dctrn_cte AS (%s) SELECT * FROM (SELECT *, ROW_NUMBER() OVER (ORDER BY (SELECT 0)) AS doctrine_rownum FROM dctrn_cte) AS doctrine_tbl WHERE doctrine_rownum BETWEEN %d AND %d ORDER BY doctrine_rownum ASC";

    $this->assertEquals(sprintf($selectFromCtePattern, $modifiedQuery, $offset + 1, $limit), $query);
}
```